src/test/java/core/basesyntax/RegistrationServiceImplTest.java

```diff
@@ -0,0 +1,93 @@
+package core.basesyntax;
+
+import core.basesyntax.db.Storage;
+import core.basesyntax.model.User;
+import core.basesyntax.service.RegistrationServiceImpl;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Test;
+
+public class RegistrationServiceImplTest {
+    private final RegistrationServiceImpl registrationService = new RegistrationServiceImpl();
+
+    @BeforeAll
+    public static void setUp() {
+        User validUser = new User();
+        validUser.setAge(18);
+        validUser.setLogin("Cherkassy");
+        validUser.setPassword("Qwerty3");
+        Storage.people.add(validUser);
+    }
+
+    @Test
+    public void register_NullOnFieldsOfUser_NotOk() {
+        User emptyUser = new User();
+        Assertions.assertThrows(RegistrationException.class, ()
+                -> registrationService.register(emptyUser));
+    }
+
+    @Test
+    public void register_ValidUser_Ok() {
+        User suitableUser = new User();
+        suitableUser.setAge(52);
+        suitableUser.setLogin("Uzhgorod");
+        suitableUser.setPassword("Pass@WorD");
+        registrationService.register(suitableUser);
+        Assertions.assertEquals(suitableUser, Storage.people.get(1));
+    }
+
+    @Test
+    public void register_DuplicateLogin_NotOk() {
+        User duplicateLogin = new User();
```

Comment:
You should create User user, then Storage.people.add(user);, then create User newUser with the same login, then assertThrows(RegistrationException.class, () -> registrationService.register(newUser));